Devscord.DiscordFramework/Middlewares/Factories/UserContextsFactory.cs

```diff
@@ -1,17 +1,19 @@
 ﻿using System.Linq;
 using Devscord.DiscordFramework.Middlewares.Contexts;
-using Devscord.DiscordFramework.Middlewares.Factories;
 using Discord;
 using Discord.WebSocket;
 
-namespace Devscord.DiscordFramework.Middlewares
+namespace Devscord.DiscordFramework.Middlewares.Factories
 {
     internal class UserContextsFactory : IContextFactory<SocketGuildUser, UserContext>
     {
         public UserContext Create(SocketGuildUser user)
         {
-            var roles = user.Roles.Select(x => new UserRole(x.Id, x.Name));
+            var userRoleFactory = new UserRoleFactory();
```

Comment:
wrzucaj to przez konstruktor 
tworzenie nowej instancji czegoś, co nie trzyma stanu (lub trzyma ale jest stały) przy każdym odpalaniu metody

to zbędne dodatkowe obliczenia, które w dodatku nie poprawiają czytelności

Reply:
Czasami psucie wydajności może być usprawiedliwione sporą poprawą czytelności lub skalowalności
a tutaj nie mamy poprawy czytelności/skalowalności, a nawet ich spadek